plaid/payment.go

```diff
@@ -27,17 +47,68 @@ type CreatePaymentRecipientResponse struct {
 	RecipientID string `json:"recipient_id"`
 }
 
-func (c *Client) CreatePaymentRecipient(
+func (c *Client) CreatePaymentRecipientWithIban(
 	name string,
 	iban string,
 	address *PaymentRecipientAddress,
 ) (resp CreatePaymentRecipientResponse, err error) {
-	jsonBody, err := json.Marshal(createPaymentRecipientRequest{
-		ClientID: c.clientID,
-		Secret:   c.secret,
-		Name:     name,
-		IBAN:     iban,
-		Address:  address,
+	jsonBody, err := json.Marshal(paymentRecipientRequestWithIban{
+		BaseCreatePaymentRecipientRequest: BaseCreatePaymentRecipientRequest{
+			ClientID: c.clientID,
+			Secret:   c.secret,
+			Name:     name,
+			Address:  address,
+		},
+		IBAN: iban,
+	})
+
+	if err != nil {
+		return resp, err
+	}
+
+	err = c.Call("/payment_initiation/recipient/create", jsonBody, &resp)
+	return resp, err
+}
+
+func (c *Client) CreatePaymentRecipientWithBacs(
+	name string,
+	address *PaymentRecipientAddress,
+	bacs *PaymentRecipientBacs,
+) (resp CreatePaymentRecipientResponse, err error) {
+	jsonBody, err := json.Marshal(paymentRecipientRequestWithBacs{
+		BaseCreatePaymentRecipientRequest: BaseCreatePaymentRecipientRequest{
+			ClientID: c.clientID,
+			Secret:   c.secret,
+			Name:     name,
+
+			Address: address,
+		},
+		BACS: bacs},
+	)
+
+	if err != nil {
+		return resp, err
+	}
+
+	err = c.Call("/payment_initiation/recipient/create", jsonBody, &resp)
+	return resp, err
+}
+
+func (c *Client) CreatePaymentRecipientWithBacsAndIban(
```

Comment:
Since we are using pointers for `PaymentRecipientAddress` and `PaymentRecipientBacs` -- we can just have one `CreatePaymentRecipient` method that handles nil values properly and does validation (including that at least one of IBAN / BACS is specified). Lmk if I'm missing something.

Reply:
yeah we could but then we'd also have to change the IBAN validation to check for an empty string. We could do that here but would prefer to have the input validation done on the server side vs doing it on the client side and panicking. I also think that having these discrete methods means these client side APIs are very explicit about what they do making it a little harder to use the incorrect methods